### ISSUE TITLE
Authentication and undefined properties error

### ISSUE THREAD

**Issue by faraz9x** [SabreDevStudio/SACS-Php#5]
Help needed...
I just downloaded the zip, enabled all the extensions and changed the auth details in  **configuration/SACSRestConfig.ini**
My API KEY was:
V1:xo30***********:DEVCENTER:EXT
And secret i found after registration with Heading of Shared Secret was rt******
So my SACSRestConfig.ini file became:
|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
environment=https://api.test.sabre.com
userId=xo30***********
group=DEVCENTER
domain=EXT
clientSecret=rt****** 
formatVersion=V1
||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||

Then I ran the form.html and submitting the three fields of origin dest and date i get these sort of errors in response:

Undefined property: stdClass::$expires_in   ** ...... stdClass::$access_token *** .... stdClass::$FareInfo

And In Shared Context Object:
.... ERR.2SG.SEC.MISSING_CREDENTIALS - Authentication data is missing
... ERR.2SG.CLIENT.INVALID_REQUEST - Request is invalid: Request payload is missing or its type is not supported

**Comment by faraz9x** [comment]
@mprajs  there?

**Comment by mprajs** [comment]
Yep, let me have a look into it.

**Comment by faraz9x** [comment]
thanks and/or for other way i also want to know how can i send the request more simply like this either from php or from js:
Like here is an "Examples: request and response" from docs:
https://api.sabre.com/v1/shop/flights?origin=ATL&destination=LAS&departuredate=2016-08-13&returndate=2016-08-15&limit=1&enabletagging=true

where to put credentials in this type of post

**Comment by mprajs** [comment]
Confirmed, but looks weird to me, because I've run the Python example, and the authentication worked fine. I got an error on the first workflow activity (404...). Looks like something is going on at the test servers. I'll investigate further later today.

As of your second question, you always put credentials into the header of the call. Have a look on the `buildHeaders()` method in the RestClient.php, which is used in the `prepareCall()` method, which builds the call.

For js example, have a look [here](https://github.com/SabreDevStudio/SACS-NodeJs) - it's a JavaScript example based on NodeJS.

**Comment by faraz9x** [comment]
fine, thanks for your time.. i will be checking it after sometime to see if it works then...

**Comment by faraz9x** [comment]
well its not getting anywhere, heres the screenshot and below is the link to my files which i have uploaded to google drive kindly view and check whats being wrong done there... i needed to set it up asap but its second day passing...
![error-shot](https://cloud.githubusercontent.com/assets/21190510/18077988/8107cb26-6ea2-11e6-8ac4-150c3b2ee16b.png)

Link to file:
[https://drive.google.com/open?id=0Bwo1iu_E1MIFeGtmOGhiVmhGT3c](url)

**Comment by mprajs** [comment]
The undefined property error is caused by an error in authentication. The token object is not being set, that's why it does not have the required properties. I'm trying to debug it right now and work with the Sabre's REST team, to get some answers. Sorry for your inconvenience.

**Comment by faraz9x** [comment]
i appreciate your response and effort, kindly inform here as soon as you find the teasing thing..

**Comment by mprajs** [comment]
Found it!
The problem was a space before a colon in the headers.
Changed from:
```php
        $headers = array(
            'Authorization : Basic '.$this->buildCredentials(),
            'Accept : */*',
            'Content-Type : application/x-www-form-urlencoded'
        );
```
to 
```php
        $headers = array(
            'Authorization: Basic '.$this->buildCredentials(),
            'Accept: */*',
            'Content-Type: application/x-www-form-urlencoded'
        );
```
You can pull the changes now :)

**Comment by mprajs** [comment]
Same goes with the RestClient.php. Fixed as well :)

**Comment by faraz9x** [comment]
great let me check it..

**Comment by faraz9x** [comment]
Yes its up and running.. much thanks @mprajs  for your quick responses...

**Comment by sahdeepak83** [comment]
this issue arise due to wrong UserId , update only first 16 digit string from here like this xo30**********  , i recenlty solved this issue in my code
V1:xo30**********:DEVCENTER:EXT

**Comment by Kuest123** [comment]
Please i need help my BFM API was working very well on Live but this morning this is the error i got 

SharedContext Object
(
    [results:SharedContext:private] => Array
        (
            [SECURITY] => 
            [origin] => LOS
            [destination] => CDG
            [departureDate] => 2018-05-28
            [cabin] => Economy
            [adt] => 1
            [cnn] => 0
            [inf] => 0
            [returnDate] => 2018-06-30
            [LeadPriceCalendar] => stdClass Object
                (
                    [status] => NotProcessed
                    [type] => Validation
                    [errorCode] => ERR.2SG.SEC.MISSING_CREDENTIALS
                    [timeStamp] => 2018-05-18T15:30:21.373-05:00
                    [message] => Authentication data is missing
                )

            [InstaFlight] => 
            [BargainFinderMax] => stdClass Object
                (
                    [status] => NotProcessed
                    [type] => Validation
                    [errorCode] => ERR.2SG.SEC.MISSING_CREDENTIALS
                    [timeStamp] => 2018-05-18T15:30:21.560-05:00
                    [message] => Authentication data is missing
Please i need help i feel very frustrated 


http://gokuest.com/new-stage/php/__EVE_21_rest_test.php
